### PR TITLE
Deployment fixes

### PR DESCRIPTION
**Pull request by okuchaiev**

### CODE REVIEW

File: jarvis/jasper_v0.8.2_chkpt_to_onnx.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import nemo
```
Comment (okuchaiev) [inline jarvis/jasper_v0.8.2_chkpt_to_onnx.py]: @ryanleary let's delete this file

File: nemo/setup.py [hunk 2]
```diff
@@ -19,7 +19,7 @@
         "License :: OSI Approved :: Apache Software License"
     ],
     install_requires=[
-        'torch>=1.2.0,<1.4',
```
Comment (borisfom) [inline nemo/setup.py]: I would just use 'torch' - that seemed to work well in terms of not breaking anything. Safer.

File: scripts/docker/launch.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+# Launch TRT JASPER container.
+
+DATA_DIR=${DATA_DIR:-"/datasets"}
+CHECKPOINT_DIR=${CHECKPOINT_DIR:-"/checkpoints"}
+RESULT_DIR=${RESULT_DIR:-"/results"}
+PROGRAM_PATH=${PROGRAM_PATH}
+    
+
+MOUNTS=""
+if [ ! -z "$DATA_DIR" ];
```
Comment (borisfom) [inline scripts/docker/launch.sh]: I will have some more changes to that, but please adopt as it is for now.

File: nemo/setup.py (hunk 2 above)
Comment (okuchaiev) [reply nemo/setup.py]: removed version check

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,20 @@
+pillow>=4.3.0
```
Comment (okuchaiev) [inline requirements.txt]: do we need this file?

File: jarvis/jasper_v0.8.2_chkpt_to_onnx.py (hunk 1 above)
Comment (ryanleary) [reply jarvis/jasper_v0.8.2_chkpt_to_onnx.py]: Yes I was going to add a flag in the other file

File: nemo/nemo/backends/pytorch/actions.py [hunk 5]
```diff
@@ -1096,14 +1097,24 @@ def __extract_dynamic_axes(port_name: str, ntype: NeuralType,
                     raise ValueError(
                         f'Example input is None, but ONNX tracing was'
                         f' attempted')
-                torch.onnx.export(module, input_example, output,
+                if output_example is None:
```
Comment (ryanleary) [inline nemo/nemo/backends/pytorch/actions.py]: There should probably be a `with torch.no_grad()` wrapping all this, no?

File: requirements.txt (hunk 4 above)
Comment (ryanleary) [reply requirements.txt]: Let's keep for now, and revisit when we finalize the container process for 20.01

File: scripts/docker/build.sh [hunk 6]
```diff
@@ -0,0 +1,4 @@
+
```
Comment (ryanleary) [inline scripts/docker/build.sh]: should delete

File: scripts/docker/launch.sh [hunk 7]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
```
Comment (ryanleary) [inline scripts/docker/launch.sh]: should delete. this is too specific/enforces more structure. i don't see much value here yet.

File: nemo/nemo/backends/pytorch/actions.py (hunk 5 above)
Comment (okuchaiev) [reply nemo/nemo/backends/pytorch/actions.py]: I'll put it inside export function - the use should not do "import framework" in examples

Comment (okuchaiev) [reply nemo/nemo/backends/pytorch/actions.py]: yes, done

File: scripts/docker/build.sh (hunk 6 above)
Comment (okuchaiev) [reply scripts/docker/build.sh]: done

File: scripts/docker/launch.sh (hunk 7 above)
Comment (okuchaiev) [reply scripts/docker/launch.sh]: deleted

File: nemo/nemo/backends/pytorch/actions.py [hunk 8]
```diff
@@ -1065,15 +1065,15 @@ def __extract_dynamic_axes(port_name: str, ntype: NeuralType,
             dynamic_axes = None
 
         local_parameters = {}
-        for key, value in module._local_parameters.items():
-            local_parameters[key] = value
+        if local_parameters is not None:
```
Comment (ryanleary) [inline nemo/nemo/backends/pytorch/actions.py]: Do you mean to check if `module._local_parameters` is not None? `local_parameters` is set immediately before as an empty dict, so it shouldn't ever be None.

File: nemo/nemo/core/neural_factory.py [hunk 9]
```diff
@@ -646,6 +646,23 @@ def deployment_export(self,
             input_example: sometimes tracing will require input examples
             output_example: Should match inference on input_example
         """
+        # Custom hacks
+        # We are checking type like this to avoid taking dependency on nemo_asr
+        if type(module).__name__ == "JasperEncoder":
```
Comment (ryanleary) [inline nemo/nemo/core/neural_factory.py]: Can we add this code to the `JasperEncoder` class somehow and have this be a universal check? Like a `prepare_for_deployment()` abstract method on TrainableNeuralModule that may or may not be implemented?

Seems less than ideal for this to be in the neural_factory code.

File: nemo/nemo/backends/pytorch/actions.py (hunk 8 above)
Comment (okuchaiev) [reply nemo/nemo/backends/pytorch/actions.py]: yes, just fixed

File: nemo/nemo/core/neural_factory.py (hunk 9 above)
Comment (okuchaiev) [reply nemo/nemo/core/neural_factory.py]: good idea, but I would say this is outside of this PR, since we'd want to do it properly

Comment (ryanleary) [reply nemo/nemo/core/neural_factory.py]: Yes, but a step in the right direction isn't much different -- 

```python
if module.hasattr("prepare_for_deployment"):
    module.prepare_for_deployment()
```

then the encoder has this code pretty much copy pasted to it, with `module` being replaced with `self`

Comment (ryanleary) [reply nemo/nemo/core/neural_factory.py]: We can build upon this in future releases, but at least that way we don't have more model-specific code in a place that should be demonstrably model-agnostic.

Comment (okuchaiev) [reply nemo/nemo/core/neural_factory.py]: let's do this properly for >0.9.0

**Pull request merged**